pages/api/zapbye.ts

```diff
@@ -0,0 +1,98 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import jsforce from 'jsforce';
+
+const CONSTANTS = {
+      SALESFORCE_EMAIL: process.env.SALESFORCE_EMAIL,
+      SALESFORCE_PASSWORD: process.env.SALESFORCE_PASSWORD,
+      SALESFORCE_SECURITY_TOKEN: process.env.SALESFORCE_SECURITY_TOKEN,
+      LEMLIST_API_KEY: process.env.LEMLIST_API_KEY,
+      SALESFORCE_TASK_SUBJECT: 'Lemlist Email',
+      SALESFORCE_LEMLISTTASK_ID: '0123k0000014TCyAAM',
+      SALESFORCE_STATUS: 'Completed',
+      SALESFORCE_TASK_SUBTYPE: 'Task',
+      SALESFORCE_SOBJECT_TYPE: 'Task',
+      LEMLIST_CAMPAIGN_FETCH_COOLDOWN_TIME_IN_MS: 30000,
+};
+
+let salesforceConnection;
+let lastCampaignCheckedTimestamp: number;
+let cachedCampaignsArray: Array<LemlistCampaign> = [];
+
+export default async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+      const {
+            method,
+      } = req;
+
+      if (method != "POST") {
+            res.status(405).json({ error: "Method Not Allowed" });
+            return;
+      }
+
+      if(!salesforceConnection || !salesforceConnection.accessToken) {
```

Comment:
I *think* you can check both in one statement. Using *optional chaning*: ([Ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining))

```suggestion
      if(!salesforceConnection?.accessToken) {
```

Reply:
Uff, gotta get used to having nullable operator in my toolkit again. Nice!